### DISCUSSION TITLE
Route validator in V5

### DISCUSSION THREAD

**Discussion by artem-itdim** [adonisjs/core#2264] [Help] [answered]
Hi there!

Is there any way in V5 to make a route level validator? The same as in V4 [Here](https://adonisjs.com/docs/4.1/validator#_route_validator)

Thanks.

**Comment by RomainLanz** [comment, marked as answer]
Hey @artem-itdim! 👋 

Short answer: **Nope**

We removed them because it doesn't fit well with TypeScript.
Instead, you can directly call the validator [from your controller](https://preview.adonisjs.com/guides/http/form-submissions#using-validator-classes), and you will have anything typed as expected.

```ts
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
import PostValidator from 'App/Validators/PostValidator'

export default class PostsController {
  public async store ({ request }: HttpContextContract) {
    const data = await request.validate(PostValidator)

    // ...
  }
}
```

**Reply by artem-itdim** [reply]
@RomainLanz, thanks a lot for your reply

**Reply by AsadJivani** [reply]
@RomainLanz can you please elaborate further on **it doesn't fit well with TypeScript**
